### ISSUE TITLE
Collection queries with old query param

### ISSUE THREAD

**Issue by thomasmassmann** [collective/collective.exportimport#31]
When exporting ATCollection items the queries might contain outdated configurations. The following query doesn’t work in Plone 5 with plone.app.querystring > 1.3.2 (https://github.com/plone/plone.app.querystring/blob/master/CHANGES.rst#1312-2015-11-26)

```json
        "query": [
            {
                "i": "portal_type", 
                "o": "plone.app.querystring.operation.selection.is", 
                "v": [
                    "News Item"
                ]
            }, 
            {
                "i": "path", 
                "o": "plone.app.querystring.operation.string.relativePath", 
                "v": "../"
            }
        ],
```

The correct version would be:

```json
        "query": [
            {
                "i": "portal_type", 
                "o": "plone.app.querystring.operation.selection.any", 
                "v": [
                    "News Item"
                ]
            }, 
            {
                "i": "path", 
                "o": "plone.app.querystring.operation.string.relativePath", 
                "v": "../"
            }
        ],
```

With `plone.app.querystring` there is an upgrade step available (8 -> 9) which you can run to fix imported collections with the wrong queries.

Besides that, should we add some logic to adjust the queries on export/import?

**Comment by pbauer** [comment]
Yes, I planned to put that in https://github.com/collective/collective.exportimport/blob/main/src/collective/exportimport/export_content.py#L318

So far I only have some half-bakes code from in a client-project.

We could add this to the  which does this:

```
        if 'query' in item and isinstance(item['query'], dict):
            # assume this is a collection-query
            item['query'] = self.fix_collection_query(item['query'], obj)

```

And `fix_collection_query` could then either call the upgrade step from plone.app.querystring or mimick what that does. A pull-request would be most welcome.

**Comment by thomasmassmann** [comment]
Sounds good to me.

**Comment by thomasmassmann** [comment]
> And `fix_collection_query` could then either call the upgrade step from plone.app.querystring or mimick what that does. A pull-request would be most welcome.

Well, if the export is done from an old site, then doing this in the export will not work, as the upgrade step from `plone.app.querystring` will not be available. Unless the upgrade step code will be copied.

An alternative would be to check this when importing the content, e.g. https://github.com/collective/collective.exportimport/blob/main/src/collective/exportimport/import_content.py#L222. Then the upgrade step would be available and we know for sure that the target system requires the adjustments.

**Comment by thomasmassmann** [comment]
Or maybe adding a new action to fix the collection criteria like https://github.com/collective/collective.exportimport/blob/main/src/collective/exportimport/import_content.py#L473
